R/evaluate-promise.R

```diff
@@ -39,9 +39,12 @@ evaluate_promise <- function(code, print = FALSE) {
   temp <- file()
   on.exit(close(temp))
 
-  result <- with_sink(temp,
-    withCallingHandlers(
-      withVisible(code), warning = wHandler, message = mHandler
+  with_reporter(
```

Comment:
Tried that. If I capture the error, I can't handle multiple warnings; if I simply handle the condition, the top-level handler still is invoked, which means that the reporter prints to the sink.